webapp/channels/src/packages/mattermost-redux/src/utils/post_utils.ts

```diff
@@ -140,16 +140,31 @@ function isJoinLeavePostForUsername(post: Post, currentUsername: string): boolea
         post.props.removedUsername === currentUsername;
 }
 
+export function isPostUploadingFile(post: Post): boolean {
+    return post.id === post.pending_post_id && post.file_client_ids !== undefined && post.file_client_ids.length !== 0;
+}
+
+export function isPostDangling(state: GlobalState, post: Post) {
+    const rootPost = state.entities.posts.posts[post.root_id];
+    return rootPost && isPostPending(rootPost);
+}
+
+export function isPostPending(post: Post) {
+    return post.id === post.pending_post_id;
+}
+
 export function isPostPendingOrFailed(post: Post): boolean {
-    return post.failed || post.id === post.pending_post_id;
+    return post.failed || isPostPending(post);
 }
 
 export function comparePosts(a: Post, b: Post): number {
+    const aIsUploadingFile = isPostUploadingFile(a);
+    const bIsUploadingFile = isPostUploadingFile(b);
     const aIsPendingOrFailed = isPostPendingOrFailed(a);
     const bIsPendingOrFailed = isPostPendingOrFailed(b);
-    if (aIsPendingOrFailed && !bIsPendingOrFailed) {
+    if (!aIsUploadingFile && aIsPendingOrFailed && !bIsPendingOrFailed) {
```

Comment:
@hmhealey
I added this condition to sort posts with uploads by their create_at.
Should I separate if-else block in this case?

Reply:
Yeah, I'm thinking of just breaking that up so that it's like
```js
if (aIsUploadingFile && !bIsUploadingFile) {
    // I might have gotten these cases backwards to what you were intending
    return -1;
} else if (!aIsUploadingFile && bIsUploadingFile) {
    return 1;
}

if (aIsPendingOrFailed && !bIsPendingOrFailed) {
    return -1;
} else if (!aIsPendingOrFailed && bIsPendingOrFailed) {
    return 1;
}

if (a.create_at > b.create_at) {
    return -1;
} else if (a.create_at < b.create_at) {
    return 1;
}

return 0;
```
I find that pattern just makes it a bit easier to read since it spreads out the different cases pretty clearly